backend/server/src/config/mod.rs

```diff
@@ -178,3 +204,50 @@ pub(crate) fn write_template(path: Option<&PathBuf>) -> Result<()> {
 
     Ok(())
 }
+
+impl Opencast {
+    fn validate(&self) -> Result<()> {
+        // We only allow HTTP if the host resolves to a loopback (local)
+        // address. We send the unencrypted `sync_password`, so HTTPS is
+        // required.
+        if self.use_insecure_connection {
+            debug!("Checking whether Opencast host '{}' is a loopback address", self.host);
+
+            let is_loopback = if let Ok(addr) = self.host.parse::<IpAddr>() {
+                addr.is_loopback()
+            } else {
+                let mut socket_addrs = if self.host.contains(':') {
```

Comment:
Because it's an iterator and `Iterator::all` takes `&mut self` instead of consuming the iterator. Which is... interesting? not what I would have expected, but okay.